cloudtrail-eventbridge-lambda-dynamodb-sam/README.md

```diff
@@ -0,0 +1,101 @@
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://gitlab.aws.dev/jrdwyer/automated-route-advertisement-removal.git
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd automated-route-advertisement-removal
+    ```
+1. From the command line, use AWS SAM to deploy the AWS resources for the workflow as specified in the template.yaml file:
+    ```
+    sam deploy --guided
+    ```
+1. During the prompts:
+    * Enter a stack name.
+    * Enter the desired AWS Region.
+    * Enter an email address for notifications when an IP address space has been advertised unexpectedly.
+    * Allow SAM CLI to create IAM roles with the required permissions.
+    * Accept all other defaults.
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+    ```
+    Stack Name [sam-app]: byoip-advertisement-removal
+    AWS Region [us-east-1]:
+    Parameter NotificationEmail []: jrdwyer@amazon.com
+    #Shows you resources changes to be deployed and require a 'Y' to initiate deploy
+    Confirm changes before deploy [y/N]: 
+    #SAM needs permission to be able to create roles to connect to the resources in your template
+    Allow SAM CLI IAM role creation [Y/n]: 
+    #Preserves the state of previously provisioned resources when an operation fails
+    Disable rollback [y/N]: 
+    Save arguments to configuration file [Y/n]: 
+    SAM configuration file [samconfig.toml]:
+    SAM configuration environment [default]:
+    ```
+
+1. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+1. The list of IP ranges that are allowed to be advertised will need to be configured by the user by creating an item in the DynamoDB table created by the template.  The item should be entered in the following format:
+```
+{
+  "id": {
+    "S": "Allowed_CIDRs"
+  },
+  "CIDRs": {
+    "L": [
+      {
+        "S": "CIDR1"
+      },
+      {
+        "S": "CIDR2"
+      },
+      {
+        "S": "CIDR3"
+      },
+      ...
+    ]
+  }
+}
+```
+
+
+## How it works
+
+Event bridge will trigger a rule any time the `AdvertiseByoipCidr` API call is logged via CloudTrail.  The lambda will look at DynamoDB to determine if the BYOIP range is permitted to be publicly advertised and withdraw the advertisement if it is not permitted. 
+
+
+## Testing
+
+In IPAM, enable advertisement of a BYOIP space that is not permitted to be advertised via the list of allowed IP ranges in DynamoDB.  If working as expected, the advertisement should be withdrawn within a few seconds after the CIDR has been advertised.
```

Comment:
Spell out any abbreviations the first time they are used (and include a link to the docs)